src/hooks.rs

```diff
@@ -0,0 +1,264 @@
+
+    // examine the UtilityStatement itself.  We're interested in three different commands,
+    // "DROP FUNCTION", "DROP SCHEMA", and "ALTER FUNCTION".  Two different node types are used
+    // for these -- `DropStmt` and `AlterFunctionStmt`
+
+    if utility_stmt.type_ == pg_sys::NodeTag_T_DropStmt {
+        let drop_stmt = PgBox::from_pg(pstmt.utilityStmt.cast::<pg_sys::DropStmt>());
+
+        // in the case of DropStmt, if the object being dropped is a FUNCTION or a SCHEMA, we'll
+        // go off and handle those two cases
+        match drop_stmt.removeType {
+            pg_sys::ObjectType_OBJECT_FUNCTION => handle_drop_function(&drop_stmt),
+            pg_sys::ObjectType_OBJECT_SCHEMA => handle_drop_schema(&drop_stmt),
+            _ => {
+                // we don't do anything for the other objects
+            }
+        }
+    } else if utility_stmt.type_ == pg_sys::NodeTag_T_AlterFunctionStmt {
+        let alter_stmt = PgBox::from_pg(pstmt.utilityStmt.cast::<pg_sys::AlterFunctionStmt>());
+
+        // and for AlterFunctioStmt, we'll just go do it.
+        handle_alter_function(&alter_stmt);
+    }
+
+    if PREVIOUS_PROCESS_UTILITY_HOOK.is_some() {
+        // previous hook must go last as we need the catalog entries this utility statement might
+        // operate on to be valid
+        let prev_hook = PREVIOUS_PROCESS_UTILITY_HOOK.as_ref().unwrap();
+        prev_hook(
+            pstmt.into_pg(),
+            query_string,
+            #[cfg(not(feature = "pg13"))]
+            read_only_tree,
+            context,
+            params,
+            query_env,
+            dest,
+            qc,
+        );
+    } else {
+        // otherwise if there isn't one, we are the first to hook ProcessUtility, so ask Postgres
+        // to do whatever it wants to do with this statement
+        pg_sys::standard_ProcessUtility(
+            pstmt.into_pg(),
+            query_string,
+            #[cfg(not(feature = "pg13"))]
+            read_only_tree,
+            context,
+            params,
+            query_env,
+            dest,
+            qc,
+        );
+    }
+}
+
+/// if the function being altered is `LANGUAGE plrust`, block any attempted change to the `STRICT`
+/// property, even if to the current value
+unsafe fn handle_alter_function(alter_stmt: &PgBox<pg_sys::AlterFunctionStmt>) {
+    let pg_proc_oid = pg_sys::LookupFuncWithArgs(alter_stmt.objtype, alter_stmt.func, false);
+    let lang_oid = lookup_func_lang(pg_proc_oid);
+
+    if lang_oid == plrust_lang_oid() {
+        // block a change to the 'STRICT' property
+        let actions = PgList::<pg_sys::DefElem>::from_pg(alter_stmt.actions);
+        for defelem in actions.iter_ptr() {
+            static STRICT: &str = "strict";
+
+            let defelem = PgBox::from_pg(defelem);
+            let name = CStr::from_ptr(defelem.defname);
+
+            if name.to_string_lossy().eq_ignore_ascii_case(STRICT) {
```

Comment:
I guess we're iterating over `defelem`s so I... are those guaranteed to be individual properties?

Reply:
I argued with myself around this.  Sadly Postgres doesn't have any kind of enum for these things.  No matter how the user spells it and no matter surrounding whitespace, Postgres' parser is always going to go for `"strict"`, but I thought a little bit of fuzziness was acceptable considering how terrible it would be for this switch to be flipped.

We could to something like `.to_lowercase().contains(STRICT)` but then Postgres 16 is gonna invent some new function probably that has the word "strict" in it that has nothing to do with this.  I mean, they probably won't, but...

Reply:
also, each "other parameter" is its own `DefElem`, so we at least have that going for us.

Reply:
> We could to something like .to_lowercase().contains(STRICT) but then Postgres 16 is gonna invent some new function probably that has the word "strict" in it that has nothing to do with this. I mean, they probably won't, but...

I am 100% okay with breaking on attempting to ALTER FUNCTION that.